.github/workflows/binary.yaml

```diff
@@ -0,0 +1,81 @@
+name: Binaries
+
+on:
+  schedule:
+    - cron: '0 1 * * *' # Triggers the build at 1:00 UTC time
+
+jobs:
+  binary:
+    name: Create
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: macos-13
+            name: x86_64-apple-darwin
+            installable: .#
+          - os: macos-14
+            name: aarch64-apple-darwin
+            installable: .#
+          - os: ubuntu-22.04
+            name: x86_64-unknown-linux-musl
+            installable: .#dune-static
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: ocaml/dune
+          ref: toolchains
```

Comment:
I think it would be nice if we could have multiple branches create nightly binaries. But it's a stretch goal.

Reply:
I wanted to start with something simple. In the future, we can expect to have something more complex.